src/effect-system/explicitInfer.ml

```diff
@@ -199,67 +175,36 @@ let stripHindleyMilnerLikeTy : Types.target_ty ->
 let instantiateVariable (x : variable) (scheme : Types.target_ty)
   : (Typed.expression * Types.target_ty * Typed.omega_ct list) =
   (* 1: Take the type signature apart *)
-  let skelVars, tyVarsWithSkels, dirtVars, tyCs, dirtCs, monotype =
+  let tyCs, dirtCs, monotype =
     (match stripHindleyMilnerLikeTy scheme with
-     | Some (a,b,c,d,e,f) -> (a,b,c,d,e,f)
+     | Some (a,b,c) -> (a,b,c)
      | None -> failwith "instantiateVariable: Non-HM type in the environment!") in
 
-  (* 2: Generate fresh skeleton, type, and dirt variables *)
-  let newSkelVars = List.map (fun _ -> CoreTypes.SkelParam.fresh ()) skelVars in
-  let newTyVars   = List.map (fun _ -> CoreTypes.TyParam.fresh ()) tyVarsWithSkels in
-  let newDirtVars = List.map (fun _ -> Types.fresh_dirt ()) dirtVars in
-
-  (* 3: Generate the freshening substitution *)
+  (* 2: Generate the freshening substitution *)
   let foldLeft f xs x0 = List.fold_left f x0 xs in (* GEORGE: Just for convenience *)
-  let sub = Substitution.empty
-            |> (* Substitute the old skeleton variables for the fresh ones *)
-               foldLeft
-                 (fun sub (oldS, newSkelVar) ->
-                    let newS = Types.SkelParam newSkelVar in
-                    sub |> Substitution.add_skel_param_substitution oldS newS)
-                 (List.combine skelVars newSkelVars)
-            |> (* Substitute the old type variables for the fresh ones *)
-               foldLeft
-                 (fun sub (oldA, newTyVar) ->
-                    let newA = Types.TyParam newTyVar in
-                    sub |> Substitution.add_type_substitution oldA newA)
-                 (List.combine (List.map fst tyVarsWithSkels) newTyVars)
-            |> (* Substitute the old dirt variables for the fresh ones *)
-               foldLeft
-                 (fun sub (oldD, newD) ->
-                    sub |> Substitution.add_dirt_substitution oldD newD)
-                 (List.combine dirtVars newDirtVars)
-  in
-
-  (* 4: Generate the wanted skeleton constraints *)
-  let wantedSkelCs = List.map (* a' : sigma(tau) *)
-                       (fun (a,s) -> Typed.TyParamHasSkel (a, subInSkel sub s))
-                       (List.combine newTyVars (List.map snd tyVarsWithSkels)) in
+  let sub = Substitution.empty in
```

Comment:
After removing dirt polymorphism too, there is no more polymorphism left at any level (type, skeleton, or dirt). This means that this substitution is not needed (that's why it ended up empty). You can delete `sub` and all its uses.